src/server/game/Spells/Spell.cpp

```diff
@@ -5992,13 +5986,13 @@ SpellCastResult Spell::CheckCast(bool strict)
                     if (res != SPELL_CAST_OK)
                         return res;
 
-                    // chance for fail at orange mining/herb/LockPicking gathering attempt
+                    // chance for fail at lockpicking attempt
                     // second check prevent fail at rechecks
                     if (skillId != SKILL_NONE && (!m_selfContainer || ((*m_selfContainer) != this)))
                     {
-                        bool canFailAtMax = skillId != SKILL_HERBALISM && skillId != SKILL_MINING;
+                        bool canFailAtMax = skillId == SKILL_LOCKPICKING;
 
-                        // chance for failure in orange gather / lockpick (gathering skill can't fail at maxskill)
+                        // chance for failure in orange lockpick
                         if ((canFailAtMax || skillValue < sWorld->GetConfigMaxSkillValue()) && reqSkillValue > irand(skillValue - 25, skillValue + 37))
```

Comment:
The condition here must be && not ||

Reply:
wait, more rework needed. I'll prepare a code snippet for it

Reply:
Here you are, this should be the correct condition for this part

```
                      if (skillId == SKILL_LOCKPICKING && reqSkillValue > irand(skillValue - 25, skillValue + 37))
                          return SPELL_FAILED_TRY_AGAIN;
```

Reply:
Also remember to remove` bool canFailAtMax = skillId == SKILL_LOCKPICKING;`

Reply:
done